src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/FindAllReferencesHandler.cs

```diff
@@ -78,35 +102,62 @@ public async Task<VSReferenceItem[]> HandleRequestAsync(ReferenceParams request,
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            var serverKind = LanguageServerKind.CSharp;
-            var referenceParams = new ReferenceParams()
+            var referenceParams = new SerializableReferenceParams()
             {
                 Position = projectionResult.Position,
                 TextDocument = new TextDocumentIdentifier()
                 {
                     Uri = projectionResult.Uri
                 },
                 Context = request.Context,
-                PartialResultToken = request.PartialResultToken
+                PartialResultToken = token // request.PartialResultToken
             };
 
-            var result = await _requestInvoker.ReinvokeRequestOnServerAsync<ReferenceParams, VSReferenceItem[]>(
+            if (!_lspProgressListener.TryListenForProgress(
+                token,
+                onProgressNotifyAsync: (value, ct) => ProcessReferenceItemsAsync(value, request.PartialResultToken, ct),
+                WaitForProgressNotificationTimeout,
+                cancellationToken,
+                out var onCompleted))
+            {
+                return null;
+            }
+
+            _ = await _requestInvoker.ReinvokeRequestOnServerAsync<SerializableReferenceParams, VSReferenceItem[]>(
                 Methods.TextDocumentReferencesName,
-                serverKind,
+                LanguageServerKind.CSharp,
                 referenceParams,
                 cancellationToken).ConfigureAwait(false);
 
-            cancellationToken.ThrowIfCancellationRequested();
+            // We must not return till we have received the progress notifications
+            // and reported the results via the PartialResultToken
+            await onCompleted.ConfigureAwait(false);
+
+            // Results returned through Progress notification
+            return Array.Empty<VSReferenceItem>();
```

Comment:
Thinking more about this, should we just capture the result of "reinvoke" and return that here? Roslyn will always return an empty array anyhow; and in the case that they again move away from IProgress<T> we'll "just work"